__init__.py

```diff
@@ -360,18 +360,40 @@ def selectLanguage( self, path ):
 		elif conf["viur.languageMethod"] == "url":
 			tmppath = urlparse.urlparse( path ).path
 			tmppath = [ urlparse.unquote( x ) for x in tmppath.lower().strip("/").split("/") ]
+			langSet = False
 			if len( tmppath )>0 and tmppath[0] in conf["viur.availableLanguages"]+list( conf["viur.languageAliasMap"].keys() ):
 				self.language = tmppath[0]
-				return( path[ len( tmppath[0])+1: ] ) #Return the path stripped by its language segment
+				langSet = True
+				return path[ len( tmppath[0])+1: ]  # Return the path stripped by its language segment
 			else: # This URL doesnt contain an language prefix, try to read it from session
 				if session.current.getLanguage():
 					self.language = session.current.getLanguage()
-				elif "X-Appengine-Country" in self.request.headers.keys():
+					langSet = True
+				if not langSet and "Accept-Language" in self.request.headers:
+					acceptLangHeader = self.request.headers["Accept-Language"]
+					if acceptLangHeader:
+						acceptLangHeader = acceptLangHeader.split(",")
+						for possibleLang in acceptLangHeader[:7]:
+							try:
```

Comment:
No need for try/except here. As you are not interested in regionEtc just call lng = possibleLang.split("-")[0] - which will always work